cmd/argo/commands/wait.go

```diff
@@ -36,17 +36,25 @@ func NewWaitCommand() *cobra.Command {
 // WaitWorkflows waits for the given workflowNames.
 func WaitWorkflows(workflowNames []string, ignoreNotFound, quiet bool) {
 	var wg sync.WaitGroup
+	wfSuccessStatus := make([]bool, 0, len(workflowNames))
```

Comment:
Minor: why not use a bool initialized to true, and set to false as soon as you see one that failed? Then you don't have to iterate this list, and simply exit 1 if the bool is false.